File: complex_ws/src/complex/scripts/complex_sub.py
#!/usr/bin/env python3

import rospy
from complex.msg import comp



def callBack(comp):
    rospy.loginfo("%.1f + %.1fj", comp.Real, comp.Imaginary)



def comp_sub():
    rospy.init_node('sub_complex', anonymous=True)
    rospy.Subscriber("complex", comp, callBack)
    rospy.spin()

if __name__ == "__main__":
    comp_sub()

File: complex_ws/src/complex/src/complex_sub.cpp
#include <ros/ros.h>
#include <complex/comp.h>

void call_Back(const complex::comp::ConstPtr& comp)
{
    ROS_INFO("%.2f + %.2fj", comp->Real, comp->Imaginary);
}

int main(int argc, char**argv)
{
    ros::init(argc,argv, "sub_complex");
    ros::NodeHandle nh;

    ros::Subscriber sub = nh.subscribe("complex", 1000, call_Back);

    ros::spin();
    return 0;

}